test/create-sys-script.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/python3
+
+OUTFILE_HEADER = u"""#!/usr/bin/python3
+#
+# create-sys-script.py
+#
+# (C) 2017 Canonical Ltd.
+# Author: Dan Streetman <dan.streetman@canonical.com>
+#
+# systemd is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# systemd is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with systemd; If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+
+# Use this only to (re-)create the test/sys-script.py script,
+# after adding or modifying anything in the test/sys/ directory
+
+
+import os, sys, stat, tempfile, filecmp
+
+
+OUTFILE = u"sys-script.py"
+OUTFILE_MODE = stat.S_IRWXU | stat.S_IRWXG | stat.S_IROTH | stat.S_IXOTH
+
+OUTFILE_FUNCS = r"""
+import os, sys
+
+def d(path, mode):
+    os.mkdir(path)
+    os.chmod(path, int(mode, 8))
+
+def l(path, src):
+    os.symlink(src, path)
+
+def f(path, mode, contents):
+    # wrap string contents with b''''''
+    s = u"b'''{}'''".format(contents)
+    # convert directly to bytes using eval
+    b = eval(s)
+    # write the bytes out to file
+    outf = open(path, "wb")
+    outf.write(b)
+    outf.close()
+    os.chmod(path, int(mode, 8))
+
+"""
+
+OUTFILE_MAIN = u"""
+if len(sys.argv) < 2:
+    print(u"Usage: {} <target dir>".format(sys.argv[0]))
+    exit(1)
+
+if not os.path.isdir(sys.argv[1]):
+    print(u"Target dir {} not found".format(sys.argv[1]))
+    exit(1)
+
+os.chdir(sys.argv[1])
+
+"""
+
+
+def mode_rwx(m):
+    return m & ( stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO )
+
+
+def handle_dir(outfile, path):
+    m = mode_rwx(os.lstat(path).st_mode)
+    outfile.write(u"d(u'{}', u'{:o}')\n".format(path, m))
```

Comment:
So here, instead of `u'{:o}'`, I'd expect `{:o}`, and then the conversion back is not needed.